bots/machine/machine_core/machine.py

```diff
@@ -264,8 +265,12 @@ def set_dns(self, nameserver=None, domain=None):
 
     def dhcp_server(self, mac='52:54:01', range=['10.111.112.2', '10.111.127.254']):
         """Sets up a DHCP server on the interface"""
-        cmd = "dnsmasq --domain=cockpit.lan --interface=\"$(grep -l '{mac}' /sys/class/net/*/address | cut -d / -f 5)\" --bind-dynamic --dhcp-range=" + ','.join(range) + " && firewall-cmd --add-service=dhcp"
-        self.execute(cmd.format(mac=mac))
+        cmd = f"dnsmasq -x {self.dhcp_server_pid_file} --domain=cockpit.lan --interface=\"$(grep -l '{mac}' /sys/class/net/*/address | cut -d / -f 5)\" --bind-dynamic --dhcp-range=" + ','.join(range) + " && firewall-cmd --add-service=dhcp"
```

Comment:
Please go back to `%` or `.format()`, to avoid problems with earlier Python (such as on our Semaphore tests).